src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.netstandard.approved.txt

```diff
@@ -750,6 +754,7 @@ namespace NServiceBus
     {
         protected internal RecoverabilityAction() { }
         public static NServiceBus.DelayedRetry DelayedRetry(System.TimeSpan timeSpan) { }
+        public static NServiceBus.Discard Discard(string reason = null) { }
```

Comment:
I have a faint memory that using optional parameters in public API is bad from a API evolving perspective since adding other overloads might break client. Can't find anything other than https://stackoverflow.com/questions/12670735/is-replacing-an-optional-parameter-with-overloads-a-breaking-change to back this up, @Particular/nservicebus-maintainers do you remember?

Ie should we expand this to:

```
public static NServiceBus.Discard Discard() { }
public static NServiceBus.Discard Discard(string reason) { }
```

Reply:
yeah pretty sure we tried to avoid optional params as those get compiled into the method invocation and can easy cause all sorts of compatibility issues when evolving the API.

Reply:
https://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/ and https://haacked.com/archive/2010/08/12/more-optional-versioning-fun.aspx/ also discuss it.